Cpp_test/lib/ipc-shm-send.cpp

```diff
@@ -98,9 +108,16 @@ void IPCShmSend::transfer()
 				throw std::runtime_error("ERROR: check lock mutex.");
 			}
 
-			if (pthread_cond_wait(&(this->shm_ptr->cond_re), &(this->shm_ptr->mutex)) != 0)
+			int retval = pthread_cond_timedwait(&(this->shm_ptr->cond_re), &(this->shm_ptr->mutex), &to);
```

Comment:
`to` is not actualised in the loop, so it will failed when all the transfer is more than 10s. That's why the shm doesn't work.